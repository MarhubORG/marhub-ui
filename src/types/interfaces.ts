import { Organization } from '../redux/actions/dashboard';
import { User } from '../redux/actions/users';

export interface RegistrationState {
  loading: boolean;
  isLoggedIn: boolean;
  error: string;
  redirect: boolean;
  loginRedirect: boolean;
  role: string;
  myOrganization: string;
}

export interface RootState {
  registration: RegistrationState;
  apiReducer: ApiState;
  dashboardReducer: DashboardState;
  userReducer: UserState;
}

export interface ApiState {
  irapState: object[];
  loading: boolean;
  error: string;
}

export interface DashboardState {
  loading: boolean;
  organizations: Organization[];
  errorMessage: string;
  redirectToVisibleFields: string;
  templateMessage: string;
}

export interface UserState {
  loading: boolean;
  users: User[];
  message: string;
}
