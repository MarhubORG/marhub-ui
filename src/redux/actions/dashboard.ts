import {
  FETCH_ORGANIZATIONS,
  FETCH_ORGANIZATIONS_FAILURE,
  FETCH_ORGANIZATIONS_SUCCESS,
  LOGOUT,
  UPDATE_ORGANIZATION,
  UPDATE_ORGANIZATION_FAILURE,
  UPDATE_ORGANIZATION_SUCCESS,
} from '../constants/actionTypes';

export interface FetchOrganizationsAction {
  type: typeof FETCH_ORGANIZATIONS;
}

export function fetchOrganizations(): FetchOrganizationsAction {
  return { type: FETCH_ORGANIZATIONS };
}

export interface OrganizationFields {
  name: string;
  visible_fields: string[];
}

export interface Organization {
  organisation: OrganizationFields;
  id: number;
}

export interface FetchOrganizationsSuccessAction {
  type: typeof FETCH_ORGANIZATIONS_SUCCESS;
  payload: Organization[];
}

export function fetchOrganizationsSuccess(
  organizations: Organization[]
): FetchOrganizationsSuccessAction {
  return { type: FETCH_ORGANIZATIONS_SUCCESS, payload: organizations };
}

export interface FetchOrganizationsFailureAction {
  type: typeof FETCH_ORGANIZATIONS_FAILURE;
}

export function fetchOrganizationsFailure(): FetchOrganizationsFailureAction {
  return { type: FETCH_ORGANIZATIONS_FAILURE };
}

export interface LogoutAction {
  type: typeof LOGOUT;
}

export function logout(): LogoutAction {
  return { type: LOGOUT };
}

export interface UpdateOrganizationPayload {
  id: number;
  organization: Organization;
}

export interface UpdateOrganizationAction {
  type: typeof UPDATE_ORGANIZATION;
  payload: UpdateOrganizationPayload;
}

export function updateOrganization(
  organization: UpdateOrganizationPayload
): UpdateOrganizationAction {
  return {
    type: UPDATE_ORGANIZATION,
    payload: organization,
  };
}

export interface UpdateOrganizationSuccessPayload {
  id: number;
  organization: Organization;
}

export interface UpdateOrganizationSuccessAction {
  type: typeof UPDATE_ORGANIZATION_SUCCESS;
  payload: UpdateOrganizationSuccessPayload;
}

export function updateOrganizationSuccess(
  organization: UpdateOrganizationSuccessPayload
): UpdateOrganizationSuccessAction {
  return {
    type: UPDATE_ORGANIZATION_SUCCESS,
    payload: organization,
  };
}

export interface UpdateOrganizationFailureAction {
  type: typeof UPDATE_ORGANIZATION_FAILURE;
  payload: string;
}

export function updateOrganizationFailure(
  message: string
): UpdateOrganizationFailureAction {
  return {
    type: UPDATE_ORGANIZATION_FAILURE,
    payload: message,
  };
}

export type DashboardActionTypes =
  | LogoutAction
  | FetchOrganizationsAction
  | FetchOrganizationsSuccessAction
  | FetchOrganizationsFailureAction
  | UpdateOrganizationAction
  | UpdateOrganizationSuccessAction
  | UpdateOrganizationFailureAction;
