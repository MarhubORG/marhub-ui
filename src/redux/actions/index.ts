import { SIGNUP, SIGNUP_REDIRECTING } from '../constants/actionTypes';
import { Action } from '../../types/interfaces';

export function signup(
  name: string,
  organization: string,
  email: string,
  password: string
): Action {
  return { type: SIGNUP, payload: { name, organization, email, password } };
}

export function signupRedirecting(): Action {
  return { type: SIGNUP_REDIRECTING };
}
