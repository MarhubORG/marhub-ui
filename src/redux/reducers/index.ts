import { combineReducers } from 'redux';
import registration from './registration';
import apiReducer from './api';
import dashboardReducer from './dashboard';

export default combineReducers({
  registration,
  apiReducer,
  dashboardReducer,
});
