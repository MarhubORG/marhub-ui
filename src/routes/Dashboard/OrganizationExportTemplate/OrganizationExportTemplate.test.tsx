import React from 'react';
import renderer from 'react-test-renderer';
import {
  UnconnectedOrganizationExportTemplate,
  mapDispatchToProps,
  MapDispatchToProps,
  mapStateToProps,
} from './OrganizationExportTemplate';
import { updateOrganization } from '../../../redux/actions/dashboard';
import { initialState } from '../../../redux/reducers/dashboard';
import { DashboardState } from '../../../types/interfaces';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <UnconnectedOrganizationExportTemplate
        updateOrganization={updateOrganization}
        organizations={[]}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

describe('mapDispatchToProps', () => {
  it('contains a function called updateOrganization', () => {
    const organization = {
      id: 1,
      organization: {
        id: 1,
        organisation: {
          visible_fields: ['age', 'current_country'],
          name: 'Org name',
        },
      },
    };
    const dispatch = jest.fn();
    const dispatches: MapDispatchToProps = mapDispatchToProps(dispatch);
    expect(Object.keys(dispatches).includes('updateOrganization')).toBeTruthy();
    expect(typeof dispatches.updateOrganization).toEqual('function');
    expect(dispatches.updateOrganization.name).toEqual('updateOrganization');
    expect(dispatches.updateOrganization(organization)).toBeUndefined();
  });
});
