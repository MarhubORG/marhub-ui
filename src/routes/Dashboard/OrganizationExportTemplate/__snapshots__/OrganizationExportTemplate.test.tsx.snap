// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c3 {
  max-height: 50vh;
}

.c2 {
  display: block;
  font-family: Open Sans,sans-serif;
  font-size: 0.85rem;
  font-weight: 700;
  line-height: 1rem;
  margin: 0.5rem 0rem;
}

.c4 {
  padding: 0.1rem 0;
}

.c0 {
  padding-left: 2rem;
}

.c1 {
  height: 2rem;
  width: 5rem;
  margin-bottom: 0.5rem;
}

<div
  className="c0"
>
  
  <h1
    className=""
  >
     Template
  </h1>
  <button
    className="c1"
    onClick={[Function]}
    type="button"
  >
    Submit
  </button>
  <form>
    <span
      className="c2"
    >
      Non-null fields
    </span>
    <div
      className="c3"
    >
      <div
        className="c4"
      >
        <label
          htmlFor="handoff_timestamp"
        >
          <input
            id="handoff_timestamp"
            name="handoff_timestamp"
            onChange={[Function]}
            type="checkbox"
            value=""
          />
          handoff_timestamp
        </label>
      </div>
    </div>
    <span
      className="c2"
    >
      Permitted fields
    </span>
    <div
      className="c3"
    >
      <div
        className="c4"
      >
        <label
          htmlFor="age"
        >
          <input
            id="age"
            name="age"
            onChange={[Function]}
            type="checkbox"
            value=""
          />
          age
        </label>
      </div>
      <div
        className="c4"
      >
        <label
          htmlFor="arcurrentcountry"
        >
          <input
            id="arcurrentcountry"
            name="arcurrentcountry"
            onChange={[Function]}
            type="checkbox"
            value=""
          />
          arcurrentcountry
        </label>
      </div>
      <div
        className="c4"
      >
        <label
          htmlFor="arfullrsc"
        >
          <input
            id="arfullrsc"
            name="arfullrsc"
            onChange={[Function]}
            type="checkbox"
            value=""
          />
          arfullrsc
        </label>
      </div>
      <div
        className="c4"
      >
        <label
          htmlFor="arnationality1"
        >
          <input
            id="arnationality1"
            name="arnationality1"
            onChange={[Function]}
            type="checkbox"
            value=""
          />
          arnationality1
        </label>
      </div>
      <div
        className="c4"
      >
        <label
          htmlFor="arrsc"
        >
          <input
            id="arrsc"
            name="arrsc"
            onChange={[Function]}
            type="checkbox"
            value=""
          />
          arrsc
        </label>
      </div>
      <div
        className="c4"
      >
        <label
          htmlFor="blocked"
        >
          <input
            id="blocked"
            name="blocked"
            onChange={[Function]}
            type="checkbox"
            value=""
          />
          blocked
        </label>
      </div>
      <div
        className="c4"
      >
        <label
          htmlFor="canadian_private_sponsorhip"
        >
          <input
            id="canadian_private_sponsorhip"
            name="canadian_private_sponsorhip"
            onChange={[Function]}
            type="checkbox"
            value=""
          />
          canadian_private_sponsorhip
        </label>
      </div>
      <div
        className="c4"
      >
        <label
          htmlFor="canadian_sponsorship_info_rating"
        >
          <input
            id="canadian_sponsorship_info_rating"
            name="canadian_sponsorship_info_rating"
            onChange={[Function]}
            type="checkbox"
            value=""
          />
          canadian_sponsorship_info_rating
        </label>
      </div>
      <div
        className="c4"
      >
        <label
          htmlFor="canadian_sponsorship_info_response"
        >
          <input
            id="canadian_sponsorship_info_response"
            name="canadian_sponsorship_info_response"
            onChange={[Function]}
            type="checkbox"
            value=""
          />
          canadian_sponsorship_info_response
        </label>
      </div>
      <div
        className="c4"
      >
        <label
          htmlFor="child_illness"
        >
          <input
            id="child_illness"
            name="child_illness"
            onChange={[Function]}
            type="checkbox"
            value=""
          />
          child_illness
        </label>
      </div>
      <div
        className="c4"
      >
        <label
          htmlFor="createdAt"
        >
          <input
            id="createdAt"
            name="createdAt"
            onChange={[Function]}
            type="checkbox"
            value=""
          />
          createdAt
        </label>
      </div>
      <div
        className="c4"
      >
        <label
          htmlFor="created_on"
        >
          <input
            id="created_on"
            name="created_on"
            onChange={[Function]}
            type="checkbox"
            value=""
          />
          created_on
        </label>
      </div>
      <div
        className="c4"
      >
        <label
          htmlFor="current_country"
        >
          <input
            id="current_country"
            name="current_country"
            onChange={[Function]}
            type="checkbox"
            value=""
          />
          current_country
        </label>
      </div>
      <div
        className="c4"
      >
        <label
          htmlFor="dapev_info_rating"
        >
          <input
            id="dapev_info_rating"
            name="dapev_info_rating"
            onChange={[Function]}
            type="checkbox"
            value=""
          />
          dapev_info_rating
        </label>
      </div>
      <div
        className="c4"
      >
        <label
          htmlFor="dapev_info_response"
        >
          <input
            id="dapev_info_response"
            name="dapev_info_response"
            onChange={[Function]}
            type="checkbox"
            value=""
          />
          dapev_info_response
        </label>
      </div>
      <div
        className="c4"
      >
        <label
          htmlFor="dapi130_info_rating"
        >
          <input
            id="dapi130_info_rating"
            name="dapi130_info_rating"
            onChange={[Function]}
            type="checkbox"
            value=""
          />
          dapi130_info_rating
        </label>
      </div>
      <div
        className="c4"
      >
        <label
          htmlFor="dapi130_info_response"
        >
          <input
            id="dapi130_info_response"
            name="dapi130_info_response"
            onChange={[Function]}
            type="checkbox"
            value=""
          />
          dapi130_info_response
        </label>
      </div>
      <div
        className="c4"
      >
        <label
          htmlFor="daprelsiv_info_rating"
        >
          <input
            id="daprelsiv_info_rating"
            name="daprelsiv_info_rating"
            onChange={[Function]}
            type="checkbox"
            value=""
          />
          daprelsiv_info_rating
        </label>
      </div>
      <div
        className="c4"
      >
        <label
          htmlFor="daprelsiv_response"
        >
          <input
            id="daprelsiv_response"
            name="daprelsiv_response"
            onChange={[Function]}
            type="checkbox"
            value=""
          />
          daprelsiv_response
        </label>
      </div>
      <div
        className="c4"
      >
        <label
          htmlFor="end_feedback_follow_up_conversation"
        >
          <input
            id="end_feedback_follow_up_conversation"
            name="end_feedback_follow_up_conversation"
            onChange={[Function]}
            type="checkbox"
            value=""
          />
          end_feedback_follow_up_conversation
        </label>
      </div>
      <div
        className="c4"
      >
        <label
          htmlFor="end_feedback_helpfulness_open_ended"
        >
          <input
            id="end_feedback_helpfulness_open_ended"
            name="end_feedback_helpfulness_open_ended"
            onChange={[Function]}
            type="checkbox"
            value=""
          />
          end_feedback_helpfulness_open_ended
        </label>
      </div>
      <div
        className="c4"
      >
        <label
          htmlFor="end_feedback_helpfulness_rating"
        >
          <input
            id="end_feedback_helpfulness_rating"
            name="end_feedback_helpfulness_rating"
            onChange={[Function]}
            type="checkbox"
            value=""
          />
          end_feedback_helpfulness_rating
        </label>
      </div>
      <div
        className="c4"
      >
        <label
          htmlFor="end_feedback_other_information_needs"
        >
          <input
            id="end_feedback_other_information_needs"
            name="end_feedback_other_information_needs"
            onChange={[Function]}
            type="checkbox"
            value=""
          />
          end_feedback_other_information_needs
        </label>
      </div>
      <div
        className="c4"
      >
        <label
          htmlFor="end_feedback_recommend_open_ended"
        >
          <input
            id="end_feedback_recommend_open_ended"
            name="end_feedback_recommend_open_ended"
            onChange={[Function]}
            type="checkbox"
            value=""
          />
          end_feedback_recommend_open_ended
        </label>
      </div>
      <div
        className="c4"
      >
        <label
          htmlFor="end_feedback_recommend_rating"
        >
          <input
            id="end_feedback_recommend_rating"
            name="end_feedback_recommend_rating"
            onChange={[Function]}
            type="checkbox"
            value=""
          />
          end_feedback_recommend_rating
        </label>
      </div>
      <div
        className="c4"
      >
        <label
          htmlFor="firstname"
        >
          <input
            id="firstname"
            name="firstname"
            onChange={[Function]}
            type="checkbox"
            value=""
          />
          firstname
        </label>
      </div>
      <div
        className="c4"
      >
        <label
          htmlFor="fullrsc"
        >
          <input
            id="fullrsc"
            name="fullrsc"
            onChange={[Function]}
            type="checkbox"
            value=""
          />
          fullrsc
        </label>
      </div>
      <div
        className="c4"
      >
        <label
          htmlFor="gender"
        >
          <input
            id="gender"
            name="gender"
            onChange={[Function]}
            type="checkbox"
            value=""
          />
          gender
        </label>
      </div>
      <div
        className="c4"
      >
        <label
          htmlFor="german_reunification_info_rating"
        >
          <input
            id="german_reunification_info_rating"
            name="german_reunification_info_rating"
            onChange={[Function]}
            type="checkbox"
            value=""
          />
          german_reunification_info_rating
        </label>
      </div>
      <div
        className="c4"
      >
        <label
          htmlFor="german_reunification_info_response"
        >
          <input
            id="german_reunification_info_response"
            name="german_reunification_info_response"
            onChange={[Function]}
            type="checkbox"
            value=""
          />
          german_reunification_info_response
        </label>
      </div>
      <div
        className="c4"
      >
        <label
          htmlFor="groups"
        >
          <input
            id="groups"
            name="groups"
            onChange={[Function]}
            type="checkbox"
            value=""
          />
          groups
        </label>
      </div>
      <div
        className="c4"
      >
        <label
          htmlFor="handoff_timestamp"
        >
          <input
            id="handoff_timestamp"
            name="handoff_timestamp"
            onChange={[Function]}
            type="checkbox"
            value=""
          />
          handoff_timestamp
        </label>
      </div>
      <div
        className="c4"
      >
        <label
          htmlFor="has_unhcr_file"
        >
          <input
            id="has_unhcr_file"
            name="has_unhcr_file"
            onChange={[Function]}
            type="checkbox"
            value=""
          />
          has_unhcr_file
        </label>
      </div>
      <div
        className="c4"
      >
        <label
          htmlFor="hired_lawyer"
        >
          <input
            id="hired_lawyer"
            name="hired_lawyer"
            onChange={[Function]}
            type="checkbox"
            value=""
          />
          hired_lawyer
        </label>
      </div>
      <div
        className="c4"
      >
        <label
          htmlFor="i130_info_rating"
        >
          <input
            id="i130_info_rating"
            name="i130_info_rating"
            onChange={[Function]}
            type="checkbox"
            value=""
          />
          i130_info_rating
        </label>
      </div>
      <div
        className="c4"
      >
        <label
          htmlFor="i130_info_response"
        >
          <input
            id="i130_info_response"
            name="i130_info_response"
            onChange={[Function]}
            type="checkbox"
            value=""
          />
          i130_info_response
        </label>
      </div>
      <div
        className="c4"
      >
        <label
          htmlFor="i730_info_rating"
        >
          <input
            id="i730_info_rating"
            name="i730_info_rating"
            onChange={[Function]}
            type="checkbox"
            value=""
          />
          i730_info_rating
        </label>
      </div>
      <div
        className="c4"
      >
        <label
          htmlFor="i730_info_response"
        >
          <input
            id="i730_info_response"
            name="i730_info_response"
            onChange={[Function]}
            type="checkbox"
            value=""
          />
          i730_info_response
        </label>
      </div>
      <div
        className="c4"
      >
        <label
          htmlFor="icorn_info_rating"
        >
          <input
            id="icorn_info_rating"
            name="icorn_info_rating"
            onChange={[Function]}
            type="checkbox"
            value=""
          />
          icorn_info_rating
        </label>
      </div>
      <div
        className="c4"
      >
        <label
          htmlFor="icorn_info_response"
        >
          <input
            id="icorn_info_response"
            name="icorn_info_response"
            onChange={[Function]}
            type="checkbox"
            value=""
          />
          icorn_info_response
        </label>
      </div>
      <div
        className="c4"
      >
        <label
          htmlFor="id"
        >
          <input
            id="id"
            name="id"
            onChange={[Function]}
            type="checkbox"
            value=""
          />
          id
        </label>
      </div>
      <div
        className="c4"
      >
        <label
          htmlFor="lack_of_durable_solutions"
        >
          <input
            id="lack_of_durable_solutions"
            name="lack_of_durable_solutions"
            onChange={[Function]}
            type="checkbox"
            value=""
          />
          lack_of_durable_solutions
        </label>
      </div>
      <div
        className="c4"
      >
        <label
          htmlFor="language"
        >
          <input
            id="language"
            name="language"
            onChange={[Function]}
            type="checkbox"
            value=""
          />
          language
        </label>
      </div>
      <div
        className="c4"
      >
        <label
          htmlFor="lgtb"
        >
          <input
            id="lgtb"
            name="lgtb"
            onChange={[Function]}
            type="checkbox"
            value=""
          />
          lgtb
        </label>
      </div>
      <div
        className="c4"
      >
        <label
          htmlFor="lppn_score"
        >
          <input
            id="lppn_score"
            name="lppn_score"
            onChange={[Function]}
            type="checkbox"
            value=""
          />
          lppn_score
        </label>
      </div>
      <div
        className="c4"
      >
        <label
          htmlFor="marital_status"
        >
          <input
            id="marital_status"
            name="marital_status"
            onChange={[Function]}
            type="checkbox"
            value=""
          />
          marital_status
        </label>
      </div>
      <div
        className="c4"
      >
        <label
          htmlFor="medical_needs_score"
        >
          <input
            id="medical_needs_score"
            name="medical_needs_score"
            onChange={[Function]}
            type="checkbox"
            value=""
          />
          medical_needs_score
        </label>
      </div>
      <div
        className="c4"
      >
        <label
          htmlFor="minor_or_adult"
        >
          <input
            id="minor_or_adult"
            name="minor_or_adult"
            onChange={[Function]}
            type="checkbox"
            value=""
          />
          minor_or_adult
        </label>
      </div>
      <div
        className="c4"
      >
        <label
          htmlFor="modified_on"
        >
          <input
            id="modified_on"
            name="modified_on"
            onChange={[Function]}
            type="checkbox"
            value=""
          />
          modified_on
        </label>
      </div>
      <div
        className="c4"
      >
        <label
          htmlFor="name"
        >
          <input
            id="name"
            name="name"
            onChange={[Function]}
            type="checkbox"
            value=""
          />
          name
        </label>
      </div>
      <div
        className="c4"
      >
        <label
          htmlFor="nationality1"
        >
          <input
            id="nationality1"
            name="nationality1"
            onChange={[Function]}
            type="checkbox"
            value=""
          />
          nationality1
        </label>
      </div>
      <div
        className="c4"
      >
        <label
          htmlFor="nationality2"
        >
          <input
            id="nationality2"
            name="nationality2"
            onChange={[Function]}
            type="checkbox"
            value=""
          />
          nationality2
        </label>
      </div>
      <div
        className="c4"
      >
        <label
          htmlFor="nationality3"
        >
          <input
            id="nationality3"
            name="nationality3"
            onChange={[Function]}
            type="checkbox"
            value=""
          />
          nationality3
        </label>
      </div>
      <div
        className="c4"
      >
        <label
          htmlFor="ongoing_usrap"
        >
          <input
            id="ongoing_usrap"
            name="ongoing_usrap"
            onChange={[Function]}
            type="checkbox"
            value=""
          />
          ongoing_usrap
        </label>
      </div>
      <div
        className="c4"
      >
        <label
          htmlFor="p3_info_rating"
        >
          <input
            id="p3_info_rating"
            name="p3_info_rating"
            onChange={[Function]}
            type="checkbox"
            value=""
          />
          p3_info_rating
        </label>
      </div>
      <div
        className="c4"
      >
        <label
          htmlFor="p3_info_response"
        >
          <input
            id="p3_info_response"
            name="p3_info_response"
            onChange={[Function]}
            type="checkbox"
            value=""
          />
          p3_info_response
        </label>
      </div>
      <div
        className="c4"
      >
        <label
          htmlFor="pause_handoffs"
        >
          <input
            id="pause_handoffs"
            name="pause_handoffs"
            onChange={[Function]}
            type="checkbox"
            value=""
          />
          pause_handoffs
        </label>
      </div>
      <div
        className="c4"
      >
        <label
          htmlFor="prescreening_complete"
        >
          <input
            id="prescreening_complete"
            name="prescreening_complete"
            onChange={[Function]}
            type="checkbox"
            value=""
          />
          prescreening_complete
        </label>
      </div>
      <div
        className="c4"
      >
        <label
          htmlFor="previous_us_resettlement"
        >
          <input
            id="previous_us_resettlement"
            name="previous_us_resettlement"
            onChange={[Function]}
            type="checkbox"
            value=""
          />
          previous_us_resettlement
        </label>
      </div>
      <div
        className="c4"
      >
        <label
          htmlFor="referral_complete"
        >
          <input
            id="referral_complete"
            name="referral_complete"
            onChange={[Function]}
            type="checkbox"
            value=""
          />
          referral_complete
        </label>
      </div>
      <div
        className="c4"
      >
        <label
          htmlFor="relative_abroad_location"
        >
          <input
            id="relative_abroad_location"
            name="relative_abroad_location"
            onChange={[Function]}
            type="checkbox"
            value=""
          />
          relative_abroad_location
        </label>
      </div>
      <div
        className="c4"
      >
        <label
          htmlFor="relative_abroad_location1"
        >
          <input
            id="relative_abroad_location1"
            name="relative_abroad_location1"
            onChange={[Function]}
            type="checkbox"
            value=""
          />
          relative_abroad_location1
        </label>
      </div>
      <div
        className="c4"
      >
        <label
          htmlFor="relative_abroad_location3"
        >
          <input
            id="relative_abroad_location3"
            name="relative_abroad_location3"
            onChange={[Function]}
            type="checkbox"
            value=""
          />
          relative_abroad_location3
        </label>
      </div>
      <div
        className="c4"
      >
        <label
          htmlFor="relative_abroad_location4"
        >
          <input
            id="relative_abroad_location4"
            name="relative_abroad_location4"
            onChange={[Function]}
            type="checkbox"
            value=""
          />
          relative_abroad_location4
        </label>
      </div>
      <div
        className="c4"
      >
        <label
          htmlFor="relative_abroad_location5"
        >
          <input
            id="relative_abroad_location5"
            name="relative_abroad_location5"
            onChange={[Function]}
            type="checkbox"
            value=""
          />
          relative_abroad_location5
        </label>
      </div>
      <div
        className="c4"
      >
        <label
          htmlFor="relative_abroad_name"
        >
          <input
            id="relative_abroad_name"
            name="relative_abroad_name"
            onChange={[Function]}
            type="checkbox"
            value=""
          />
          relative_abroad_name
        </label>
      </div>
      <div
        className="c4"
      >
        <label
          htmlFor="relative_abroad_name1"
        >
          <input
            id="relative_abroad_name1"
            name="relative_abroad_name1"
            onChange={[Function]}
            type="checkbox"
            value=""
          />
          relative_abroad_name1
        </label>
      </div>
      <div
        className="c4"
      >
        <label
          htmlFor="relative_abroad_name2"
        >
          <input
            id="relative_abroad_name2"
            name="relative_abroad_name2"
            onChange={[Function]}
            type="checkbox"
            value=""
          />
          relative_abroad_name2
        </label>
      </div>
      <div
        className="c4"
      >
        <label
          htmlFor="resettlement_relatives"
        >
          <input
            id="resettlement_relatives"
            name="resettlement_relatives"
            onChange={[Function]}
            type="checkbox"
            value=""
          />
          resettlement_relatives
        </label>
      </div>
      <div
        className="c4"
      >
        <label
          htmlFor="rsc"
        >
          <input
            id="rsc"
            name="rsc"
            onChange={[Function]}
            type="checkbox"
            value=""
          />
          rsc
        </label>
      </div>
      <div
        className="c4"
      >
        <label
          htmlFor="rsc_case_status_portal"
        >
          <input
            id="rsc_case_status_portal"
            name="rsc_case_status_portal"
            onChange={[Function]}
            type="checkbox"
            value=""
          />
          rsc_case_status_portal
        </label>
      </div>
      <div
        className="c4"
      >
        <label
          htmlFor="rsc_contact_email"
        >
          <input
            id="rsc_contact_email"
            name="rsc_contact_email"
            onChange={[Function]}
            type="checkbox"
            value=""
          />
          rsc_contact_email
        </label>
      </div>
      <div
        className="c4"
      >
        <label
          htmlFor="stopped"
        >
          <input
            id="stopped"
            name="stopped"
            onChange={[Function]}
            type="checkbox"
            value=""
          />
          stopped
        </label>
      </div>
      <div
        className="c4"
      >
        <label
          htmlFor="svt"
        >
          <input
            id="svt"
            name="svt"
            onChange={[Function]}
            type="checkbox"
            value=""
          />
          svt
        </label>
      </div>
      <div
        className="c4"
      >
        <label
          htmlFor="textitContactName"
        >
          <input
            id="textitContactName"
            name="textitContactName"
            onChange={[Function]}
            type="checkbox"
            value=""
          />
          textitContactName
        </label>
      </div>
      <div
        className="c4"
      >
        <label
          htmlFor="textitContactUuid"
        >
          <input
            id="textitContactUuid"
            name="textitContactUuid"
            onChange={[Function]}
            type="checkbox"
            value=""
          />
          textitContactUuid
        </label>
      </div>
      <div
        className="c4"
      >
        <label
          htmlFor="textitSessionId"
        >
          <input
            id="textitSessionId"
            name="textitSessionId"
            onChange={[Function]}
            type="checkbox"
            value=""
          />
          textitSessionId
        </label>
      </div>
      <div
        className="c4"
      >
        <label
          htmlFor="unaccompanied_minor"
        >
          <input
            id="unaccompanied_minor"
            name="unaccompanied_minor"
            onChange={[Function]}
            type="checkbox"
            value=""
          />
          unaccompanied_minor
        </label>
      </div>
      <div
        className="c4"
      >
        <label
          htmlFor="unique_session_id"
        >
          <input
            id="unique_session_id"
            name="unique_session_id"
            onChange={[Function]}
            type="checkbox"
            value=""
          />
          unique_session_id
        </label>
      </div>
      <div
        className="c4"
      >
        <label
          htmlFor="updatedAt"
        >
          <input
            id="updatedAt"
            name="updatedAt"
            onChange={[Function]}
            type="checkbox"
            value=""
          />
          updatedAt
        </label>
      </div>
      <div
        className="c4"
      >
        <label
          htmlFor="urns"
        >
          <input
            id="urns"
            name="urns"
            onChange={[Function]}
            type="checkbox"
            value=""
          />
          urns
        </label>
      </div>
      <div
        className="c4"
      >
        <label
          htmlFor="us_relative_relationship"
        >
          <input
            id="us_relative_relationship"
            name="us_relative_relationship"
            onChange={[Function]}
            type="checkbox"
            value=""
          />
          us_relative_relationship
        </label>
      </div>
      <div
        className="c4"
      >
        <label
          htmlFor="us_relative_relationship1"
        >
          <input
            id="us_relative_relationship1"
            name="us_relative_relationship1"
            onChange={[Function]}
            type="checkbox"
            value=""
          />
          us_relative_relationship1
        </label>
      </div>
      <div
        className="c4"
      >
        <label
          htmlFor="usrap_info_rating"
        >
          <input
            id="usrap_info_rating"
            name="usrap_info_rating"
            onChange={[Function]}
            type="checkbox"
            value=""
          />
          usrap_info_rating
        </label>
      </div>
      <div
        className="c4"
      >
        <label
          htmlFor="usrap_info_response"
        >
          <input
            id="usrap_info_response"
            name="usrap_info_response"
            onChange={[Function]}
            type="checkbox"
            value=""
          />
          usrap_info_response
        </label>
      </div>
      <div
        className="c4"
      >
        <label
          htmlFor="uuid"
        >
          <input
            id="uuid"
            name="uuid"
            onChange={[Function]}
            type="checkbox"
            value=""
          />
          uuid
        </label>
      </div>
      <div
        className="c4"
      >
        <label
          htmlFor="vulnerability_assessed"
        >
          <input
            id="vulnerability_assessed"
            name="vulnerability_assessed"
            onChange={[Function]}
            type="checkbox"
            value=""
          />
          vulnerability_assessed
        </label>
      </div>
      <div
        className="c4"
      >
        <label
          htmlFor="western_left"
        >
          <input
            id="western_left"
            name="western_left"
            onChange={[Function]}
            type="checkbox"
            value=""
          />
          western_left
        </label>
      </div>
      <div
        className="c4"
      >
        <label
          htmlFor="western_right"
        >
          <input
            id="western_right"
            name="western_right"
            onChange={[Function]}
            type="checkbox"
            value=""
          />
          western_right
        </label>
      </div>
      <div
        className="c4"
      >
        <label
          htmlFor="women_and_girls_at_risk"
        >
          <input
            id="women_and_girls_at_risk"
            name="women_and_girls_at_risk"
            onChange={[Function]}
            type="checkbox"
            value=""
          />
          women_and_girls_at_risk
        </label>
      </div>
    </div>
    <br />
  </form>
</div>
`;
